mesonbuild/build.py

```diff
@@ -451,15 +451,22 @@ def get_outputs(self, backend: 'Backend') -> T.List[str]:
             for source in self.get_sources(self.srclist, self.genlist)
         ]
 
+EnvInitValueType = T.Dict[str, T.Union[str, T.List[str]]]
+
 class EnvironmentVariables(HoldableObject):
-    def __init__(self, values: T.Optional[T.Dict[str, str]] = None) -> None:
+    def __init__(self, values: T.Optional[EnvValueType] = None,
+                 init_method: str = 'set', separator: str = os.pathsep) -> None:
         self.envvars: T.List[T.Tuple[T.Callable[[T.Dict[str, str], str, T.List[str], str], str], str, T.List[str], str]] = []
         # The set of all env vars we have operations for. Only used for self.has_name()
         self.varnames: T.Set[str] = set()
 
         if values:
+            methods = {'set': self.set,
+                       'prepend': self.prepend,
+                       'append': self.append}
+            init = methods[init_method]
             for name, value in values.items():
-                self.set(name, [value])
+                init(name, listify(value), separator)
```

Comment:
```suggestion
                getattr(self, 'init_method')(name, listify(value), separator)
```

Then  you can get rid of the `methods` dict`

Reply:
oh nice trick using getattr(). However, I changed to only use it once before the loop.